versions.go

```diff
@@ -1,26 +1,66 @@
 /*
 MongoDB Versions
 
-The MongoDBVersion type provides support for interacting with MongoDB
+The LegacyMongoDBVersion type provides support for interacting with MongoDB
 versions. This type makes it possible to validate MongoDB version
 numbers and ask common questions about MongoDB versions.
 */
 package bond
 
 import (
 	"fmt"
+	"github.com/pkg/errors"
 	"sort"
 	"strconv"
 	"strings"
 
 	"github.com/blang/semver"
 )
 
-// MongoDBVersion is a structure representing a version identifier for
-// MongoDB. Use the associated methods to ask questions about MongoDB
+const endOfLegacy = "4.5.0"
+
+// MongoDBVersion encapsulates information about a MongoDBVersion.
+// Use the associated methods to ask questions about MongoDB
 // versions. All parsing of versions happens during construction, and
-// individual method calls are very light-weight.
-type MongoDBVersion struct {
+// individual method calls are very light-weight. Note that
+// not all methods are applicable for all versions.
+type MongoDBVersion interface {
+	// String returns a string representation of the MongoDB version number.
+	String() string
+	// Parsed returns the parsed version object for the version
+	Parsed() semver.Version
+	// Series returns the release series for legacy versions.
+	Series() string
+	// IsReleaseCandidate returns true if the version is a release candidate.
```

Comment:
For the legacy version funcs, can you specify that `false` is returned if they it is not legacy.

Reply:
I would prefer to leave the legacy documentation to the LegacyMongoDBVersion docstring. Putting implementation details in the interface seems wrong.

Reply:
I'll change it to "legacy version." I do think implementation details shouldn't be in the interface, but at that high level it still seems like defining what the function applies to.